### ISSUE TITLE
Allow setting access-control-allow-origin

### ISSUE THREAD

**Issue by henrahmagix** [stoplightio/prism#791]
**User story.**
As a developer using Prism with a browser app, I can set the value of `access-control-allow-origin` to something other than `*`, so that I can use Prism to stub the services my browser app calls and the browser does not block access to the response.

**Is your feature request related to a problem?**
I cannot change my browser app (written with Angular) because I need to send cookies for authentication, which is done via [`withCredentials: true`](https://angular.io/api/common/http/HttpRequest#withCredentials), and that seems to be setting the credentials mode to `include`.

**Describe the solution you'd like**
I think an option like `--cors-origin="http://localhost"` would work to start with.

It would also be great to tell the CORS plugin to set the value to the origin of the request, perhaps like `--cors-origin-match-request`? For me, this would override the need for the previous solution.

**Additional context**

Console error shows me that I need to set `access-control-allow-origin: http://localhost` to allow the browser access to the Prism response:
>Access to XMLHttpRequest at 'http://localhost:4010/foo' from origin 'http://localhost:4200' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.

I love this repo, and I'm _so close_ to getting consumer-driven contract testing working with API descriptions using Prism! This is the last hurdle I think =)

**Comment by XVincentX** [comment]
@henrahmagix PR created to fix such issue. Could you try it out and let me know if that solves your issue?

**Comment by henrahmagix** [comment]
Wow that was quick, thanks @XVincentX!

I tried to clone this repo and run `docker build .` on that branch but had an issue: the image built ok, but when it received a request it gave me an error:
>Request terminated with error: TypeError: prismInstance.process is not a function

So instead I ran a shell in stoplight/prism:3 (version 3.1.1) and made the same change, but my browser is still telling me the response is `access-control-allow-origin: *`. This is what I ran:
```bash
$ docker run --rm -it -p 4010:4010 -v $PWD/openapi.yaml:/openapi.yaml --entrypoint=sh stoplight/prism:3
/usr/src/prism/packages/cli # sed -i 's/\(server.register(fastifyCors\)/\1, {origin:true}/' ../http-server/dist/server.js
/usr/src/prism/packages/cli # grep server.register ../http-server/dist/server.js
        server.register(fastifyCors, {origin:true});
/usr/src/prism/packages/cli # node dist/index.js mock -h 0.0.0.0 /openapi.yaml
# the prism server runs and responds to my request
```

Am I testing this correctly?

**Comment by XVincentX** [comment]
I would suggest you to not build the images locally because of an internal limitation we'll hopefully get rid of very soon. In meantime — compile the code and create binary out of it

```
yarn build
yarn build.binary
./cli-binaries/prism mock …
```

**Comment by XVincentX** [comment]
@henrahmagix #800 should fix that.

**Comment by henrahmagix** [comment]
@XVincentX That's worked, thanks! I can see `access-control-allow-origin: http://localhost:4200` in the response.

However, there's still a CORS problem: the `Access-Control-Allow-Credentials` header needs to be set to 'true'. I'm not sure if that can always be set to true, or if it should only be set when the credentials mode is 'include'.

Here's my full error:
>Access to XMLHttpRequest at 'http://localhost:4010/foo' from origin 'http://localhost:4200' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Credentials' header in the response is '' which must be 'true' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.

I'm happy for that to be a separate issue, since this one is satisfied by #797 =)

**Comment by XVincentX** [comment]
Added the header in the same PR.

**Comment by henrahmagix** [comment]
@XVincentX Yup, that worked! Everything works on my end with #797 =) Thanks for responding so fast!

**Comment by henrahmagix** [comment]
Hi @XVincentX, do you know when we might get this fix into a release?

**Comment by XVincentX** [comment]
@henrahmagix Prism 3.2.0's been released.
